enhancements/etcd/etcd-size-tuning.md

```diff
@@ -0,0 +1,192 @@
+* Defragmentation, and its effects, will need to be tested as the API server will be unresponsive during this time; we will need to understand the general magnitude of defrag time and documentation those expectations.
+* Downgrades may be impacted if:
+  1. The downgrade is to a version without this feature.
+  2. The quota has been set higher than the default.
+  3. The etcd backends are using more space than the default.
+  A mitigation would be to keep the quota at the higher value (with no way of decreasing it), so that the cluster will continue to function and no data will be lost.
+
+### Drawbacks
+
+* There will be a required etcd rollout when changing values, this happens automatically by the Cluster Etcd Operator to re-render the etcd podspec.
+
+## Design Details
+
+### Open Questions [optional]
+
+For each of these, we would need to compare against the current default of 8GiB. Increasing the quota does not necessarily increase the following, but at the extreme case (e.g. increase to 16GiB and using nearly that much) we expect the following to perform differently.
+* How much memory do we require for a given database size?
+* How long does it take to sync a snapshot of a given size to another peer; for reboot/recovery?
+* How long does it take to compact & defrag a database of a given size?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+We will need to test a cluster with a larger quota to ensure everything works as expected.
+We should also load the database with a larger quota and make note of the memory requirements of a defragmentation.
+  - We should do this with some values we could assume a customer would use; 16GiB, 32GiB, 100GiB (in the extreme).
+We need to test that it is not possible to shrink the quota once it has be set to a larger value.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+None
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+- Customer feedback
+- API extension
+- Value change with minimal disruption
+- Performance testing for some larger-than-default values
+- End user documentation
+  - Description of the feature and expectations of memory usages for some largers values.
+  - Steps to change the value.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+- If the user attempts to set the quota to an invalid value (not in the form of a resource quantity (e.g. 8GiB, etc)), the API will not accept the value and return an error.
+- If the user attempts to set the quota to a lower value than the current (e.g. 21GiB -> 16GiB), the API will not accept the value and return an error.
+
+## Implementation History
+
+None
+
+## Alternatives
+
+### Discrete Profile values
```

Comment:
This seems like an interesting idea, why was it rejected?

Reply:
https://github.com/openshift/enhancements/pull/1549#discussion_r1465215686

Having a fully configurable limit lets customers better fit the limit to their own available memory.
The actual memory use will also be anything up to the limit (with some overhead for other things like the WAL file).

As long as the max limit is safe to use, discretizing the allowed limits in between (e.g in steps 16 GB, 32 GB, 64GB) provides no benefit in terms of performance or safety of the etcd cluster. Maybe a simpler API with enums but it trades off better utilization of the available resources.

Reply:
It would be good to capture that in the doc.